src/component/admin/apiToken/ApiTokenForm/SelectProjectInput.tsx

```diff
@@ -0,0 +1,150 @@
+    options,
+    defaultValue = [ALL_PROJECTS],
+    onChange,
+    disabled,
+    error,
+    onFocus,
+}) => {
+    const styles = useStyles();
+    const [projects, setProjects] = useState<string[]>(
+        typeof defaultValue === 'string' ? [defaultValue] : defaultValue
+    );
+    const [isWildcardSelected, selectWildcard] = useState(
+        typeof defaultValue === 'string' || defaultValue.includes(ALL_PROJECTS)
+    );
+    const selectAllButton = useMemo(() => {
+        const isAllSelected = projects.length === options.length;
+        return (
+            <Box sx={{ ml: 3.5, my: 0.5 }}>
+                <Link
+                    onClick={() => {
+                        setProjects(
+                            isAllSelected
+                                ? []
+                                : options.map(({ value }) => value)
+                        );
+                    }}
+                    className={styles.selectOptionsLink}
+                >
+                    {isAllSelected ? 'Deselect all' : 'Select all'}
+                </Link>
+            </Box>
+        );
+    }, [projects.length, options, styles.selectOptionsLink]);
+
+    return (
+        <Box sx={{ mt: -1, mb: 3 }}>
+            <Box sx={{ mt: 1, mb: 0.25, ml: 1.5 }}>
+                <FormControlLabel
+                    disabled={disabled}
+                    control={
+                        <Checkbox
+                            checked={disabled || isWildcardSelected}
+                            onChange={e => {
+                                if (e.target.checked) {
+                                    selectWildcard(true);
+                                    onChange([ALL_PROJECTS]);
+                                } else {
+                                    selectWildcard(false);
+                                    onChange(
+                                        projects.includes(ALL_PROJECTS)
+                                            ? []
+                                            : projects
+                                    );
+                                }
+                            }}
+                        />
+                    }
+                    label="ALL current and future projects"
+                />
+            </Box>
+            <Autocomplete
+                disabled={disabled || isWildcardSelected}
+                multiple
+                limitTags={2}
+                options={options}
+                disableCloseOnSelect
+                getOptionLabel={({ label }: IAutocompleteBoxOption) => label}
+                groupBy={() => 'Select/Deselect all'}
+                renderGroup={({ key, children }) => (
+                    <Fragment key={key}>
+                        {selectAllButton}
+                        {children}
+                    </Fragment>
+                )}
+                fullWidth
+                PaperComponent={CustomPaper}
+                renderOption={(
```

Comment:
I'd move these render fns into consts as well, just to make the whole return a bit shorter.